src/gpu/amd/miopen_softmax_impl.hpp

```diff
@@ -68,8 +70,12 @@ struct miopen_softmax_fwd_impl_t : public miopen_softmax_impl_base_t {
 
         if (pd->has_zero_dim_memory()) return status::success;
 
-        if (pd->ndims() > MIOPEN_DIM_MAX) { return status::invalid_arguments; }
-        ndims = pd->ndims() < 4 ? 4 : pd->ndims();
+        if (pd->ndims() > miopen_softmax_max_ndims) {
+            return status::invalid_arguments;
+        }
+        ndims = pd->ndims() < miopen_softmax_max_ndims
+                ? miopen_softmax_max_ndims
+                : pd->ndims();
```

Comment:
Why keeping this logic if the functionality is only designed to work on `ndims = 4`?

The comment related to why `MIOPEN_DIM_MAX` is not working is missing.

Reply:
Removing this logic and adding a comment.